ctapipe/image/pixel_likelihood.py

```diff
@@ -51,214 +46,152 @@ class PixelLikelihoodError(RuntimeError):
+def neg_log_likelihood(
+    image, prediction, spe_width, pedestal, prediction_safety=20.0,
 ):
     """
     Safe implementation of the poissonian likelihood implementation,
     adaptively switches between the full solution and the gaussian
-    approx depending on the signal. Pedestal safety parameter 
-    determines cross over point between the two solutions,
-    based on the expected p.e. resolution of the image pixels.
-    Therefore the cross over point will change dependent on 
-    the single p.e. resolution and pedestal levels.
+    approx depending on the prediction. Prediction safety parameter
+    determines cross over point between the two solutions.
 
     Parameters
     ----------
     image: ndarray
-        Pixel amplitudes from image
+        Pixel amplitudes from image (:math:`s`).
     prediction: ndarray
-        Predicted pixel amplitudes from model
+        Predicted pixel amplitudes from model (:math:`μ`).
     spe_width: ndarray
-        width of single p.e. distribution
-    ped: ndarray
-        width of pedestal
-    pedestal_safety: float
-        Decision point to choose between poissonian likelihood 
-        and gaussian approximation (p.e. resolution)
-    width_fac: float
-        Factor to determine range of summation on integral
-    dtype: datatype
-        Data type of output array
+        Width of single p.e. peak (:math:`σ_γ`).
+    pedestal: ndarray
+        Width of pedestal (:math:`σ_p`).
+    prediction_safety: float
+        Decision point to choose between poissonian likelihood
+        and gaussian approximation.
 
     Returns
     -------
-    ndarray: pixel likelihoods
+    float
     """
-    # Convert everything to arrays to begin
-    image = np.asarray(image, dtype=dtype)
-    prediction = np.asarray(prediction, dtype=dtype)
-    spe_width = np.asarray(spe_width, dtype=dtype)
-    ped = np.asarray(ped, dtype=dtype)
 
-    # Calculate photoelectron resolution
+    approx_mask = prediction > prediction_safety
 
-    width = ped * ped + image * spe_width * spe_width
-    width = np.asarray(width)
-    width[width < 0] = 0  # Set width to 0 for negative pixel amplitudes
-    width = np.sqrt(width)
-
-    like = np.zeros(image.shape)
-    # If larger than safety value use gaussian approx
-    poisson_pix = width <= pedestal_safety
-    gaus_pix = width > pedestal_safety
-
-    if np.any(poisson_pix):
-        like[poisson_pix] = poisson_likelihood_full(
-            image[poisson_pix],
-            prediction[poisson_pix],
-            spe_width,
-            ped,
-            width_fac,
-            dtype,
+    neg_log_l = 0
+    if np.sum(approx_mask) > 0:
+        neg_log_l += neg_log_likelihood_approx(
+            image[approx_mask], prediction[approx_mask], spe_width, pedestal,
         )
-    if np.any(gaus_pix):
-        like[gaus_pix] = poisson_likelihood_gaussian(
-            image[gaus_pix], prediction[gaus_pix], spe_width, ped
+
+    if np.sum(~approx_mask) > 0:
```

Comment:
This is `not np.any(...)`, right?

Reply:
`not np.all(...)` I guess .. will check if there are speed differences between this one and `np.any(~...)`

Reply:
well, there is a difference: 
```python
In [24]: mask = np.random.randint(0, 2, size=2000).astype('bool')

In [25]: %timeit np.any(~mask)
4.28 µs ± 22.8 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [26]: %timeit not np.all(mask)
3.23 µs ± 19.3 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

but am inclined to keep `np.any(~approx_mask)` due to lower mental overhead

Reply:
I dont see why having the tilde as unary binary not inside is clearer than not all

Reply:
I added the `if` clauses to not give an empty array into the function. I think `any -> not empty (regarding Trues)` is clear. But  `not all -> not empty (regarding Falses)` seems harder than `any (of the other) -> not empty`. Maybe it's just me, I have trouble putting this into written words

Reply:
This is "hot code" which will be called often by a minimizer, if there is not a huge difference in readability, we should choose the faster code.